processor/spanmetricsprocessor/factory_test.go

```diff
@@ -42,12 +44,12 @@ func TestNewProcessor(t *testing.T) {
 			name:                    "1 configured latency histogram bucket should result in 1 explicit latency bucket (+1 implicit +Inf bucket)",
 			latencyHistogramBuckets: []time.Duration{2 * time.Millisecond},
 			dimensions: []Dimension{
-				{"http.method", &defaultMethod},
-				{"http.status_code", nil},
+				{Name: "http.method", Default: &defaultMethod},
+				{Name: "http.status_code"},
 			},
 			wantLatencyHistogramBuckets: []float64{2},
-			wantDimensions: []Dimension{
-				{"http.method", &defaultMethod},
+			wantDimensions: []dimension{
+				{name: "http.method", value: &defaultMethodValue},
 				{"http.status_code", nil},
```

Comment:
nit: probably should do it for both if we want to
```suggestion
				{name: "http.status_code", value: nil},
```